irohad/ametsuchi/impl/flat_file/flat_file.cpp

```diff
@@ -160,3 +161,16 @@ FlatFile::FlatFile(std::string path,
     : dump_dir_(std::move(path)),
       available_blocks_(std::move(existing_files)),
       log_{std::move(log)} {}
+
+void FlatFile::reload() const {
+  available_blocks_.clear();
+  for (auto it = boost::filesystem::directory_iterator{dump_dir_};
+       it != boost::filesystem::directory_iterator{};
+       ++it) {
+    if (auto id = FlatFile::name_to_id(it->path().filename().string())) {
+      available_blocks_.insert(*id);
+    } else {
+      boost::filesystem::remove(it->path());
+    }
```

Comment:
You don't need to clear this cache and iterate through the whole block store on each call. You know the size of cache and you expect the exact file with determined name.